lib/facter/consul.rb

```diff
@@ -0,0 +1,24 @@
+require 'facter'
+
+# Example:
+# $ consul -v
+# Consul v0.5.0
+# Consul Protocol: 2 (Understands back to: 1)
+consul_v = Facter::Util::Resolution.exec("consul -v")
```

Comment:
This seems like it could be potentially brittle or hard to refactor.

Can you add a test for this to stub out exec with the example that your provided in the comments and assert the desired expected output?